crates/fluvio-connector-common/src/monitoring.rs

```diff
@@ -44,29 +44,37 @@ async fn start_monitoring(metrics: Arc<ConnectorMetrics>) -> Result<(), IoError>
         }
     };
 
-    // check if file exists
-    if let Ok(_metadata) = std::fs::metadata(&metric_out_path) {
-        info!("metric file already exists, deleting: {}", metric_out_path);
-        match std::fs::remove_file(&metric_out_path) {
-            Ok(_) => {}
-            Err(err) => {
-                println!("error deleting metric file: {err}");
-                return Err(err);
+    loop {
```

Comment:
This is a very risky construction. Any mistake can lead to an endless loop, which will eat all CPU. Having here counted loop or some delay after an iteration could reduce risks.

Reply:
I have a fixed 5 seconds sleep after each loop, but right, I will change to a Exponential backoff approach

Reply:
I didn't notice one removed curly bracket :). I think fixed delays are enough here.